base/pmap.jl

```diff
@@ -15,38 +15,81 @@ Note that `f` must be made available to all worker processes; see
 and Loading Packages <man-parallel-computing-code-availability>`)
 for details.
 """
-function pgenerate(p::WorkerPool, f, c)
-    if length(p) == 0
-        return AsyncGenerator(f, c)
+function pgenerate(p::WorkerPool, f, c; config=DEFAULT_PMAP_ARGS)
+    batch_size = config[:batch_size]
+    on_error = config[:on_error]
+    distributed = config[:distributed]
+
+    if  (distributed == false) ||
+        (length(p) == 0) ||
+        (length(p) == 1 && fetch(p.channel) == myid())
+
+        return AsyncGenerator(f, c; on_error=on_error)
+    end
+
```

Comment:
At this point there should be a short-cut to avoid unnecessary batching and flattening...
```julia
if batch_size == 1
    return AsyncGenerator(remote(f), c; on_error=on_error)
end
```